hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/util/HoodieSparkRecordUtils.java

```diff
@@ -91,21 +60,15 @@ private static Option<String> getNullableValAsString(StructType structType, Inte
    * @param structType  {@link StructType} instance.
    * @return Column value if a single column, or concatenated String values by comma.
    */
-  public static Object getRecordColumnValues(InternalRow row,
+  public static ComparableList getRecordColumnValues(InternalRow row,
```

Comment:
Same comment as above -- we should not be coercing this to Comparable and keep it generic